container-desktop/ContainerDesktop/Services/DnsConfigurator.cs

```diff
@@ -104,6 +105,35 @@ private void ConfigureResolvConf()
         }
     }
 
+    private void ConfigureCustomHostFile()
+    {
+        var hostFileContent = new StringBuilder();
+        var ipAddress = _configurationService.Configuration.HostEntryMode switch
+        {
+            HostEntryMode.Wsl => GetWslDnsAddresses().FirstOrDefault(),
+            HostEntryMode.Auto => GetPrimaryAdapterAddress(),
+            HostEntryMode.Static => GetAdapterAddress(_configurationService.Configuration.HostEntryAdapter),
+            _ => null
+        };
+        if (ipAddress == null)
+        {
+            _logger.LogWarning("Could not resolve  IP addresses for Host Entry Mode={hostEntryMode}", _configurationService.Configuration.HostEntryMode);
+        }
+        else
+        {
+            hostFileContent.Append($"{ipAddress}\thost.docker.internal\n");

```

Comment:
@markrexwinkel  do we only want to use the KNONW docker names or also apply our own entries